### PR TITLE
basic linting rules

### PR DESCRIPTION
**Pull request by rymaju**
Added some basic linting for our typescript react files.

Its mostly just extensions of recommended rules. In summary:

- Single quotes for non JSX
- Double quotes for JSX
- Semicolons

The rest is mostly common sense linting I don't think should be controversial.

I set up two commands:
- `npm run lint` which lints without trying to fix
- `npm run fix` which lints and fixes thing automatically if possible

Ideally we would just be able to do `npm run lint --fix` but I haven't figured out how to do that... if anybody knows how that would be very appreciated.

Take a look inside App.jsx for an example of how it looks:

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,24 +26,26 @@
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test",
-    "eject": "react-scripts eject"
-  },
-  "eslintConfig": {
-    "extends": "react-app"
+    "eject": "react-scripts eject",
```
Comment (jackblanc) [inline package.json:29]: rule suggestion: trailing commas would prevent this line from being added to the diff

Comment (jackblanc) [reply package.json:29]: ah this file might be ignored

**Review by jackblanc — APPROVED** [review]
LGTM

File: package.json (hunk 1 above)
Comment (rymaju) [reply package.json:29]: Yep it only lints .tsx, I think thats ok since we won't really be messing with the package.json too much.

**Pull request merged**